src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -246,86 +251,75 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
+      this._small = true;
+    } else {
+      this._medium = true;
     }
 
-    (this.shadowRoot!.querySelector(
-      "#thermostat"
-    ) as HTMLElement)!.style.height = radius * 2 + "px";
-
-    const loaded = await loadRoundslider();
-
-    this._roundSliderStyle = loaded.roundSliderStyle;
-    this._jQuery = loaded.jQuery;
+    await this.updateComplete;
+    this._loaded = true;
 
-    const [sliderValue, uiValue, sliderType] = this._genSliderValue(stateObj);
+    const uiValue = this._getSetTemp(stateObj);
 
-    this._jQuery("#thermostat", this.shadowRoot).roundSlider({
-      ...thermostatConfig,
-      radius,
-      min: stateObj.attributes.min_temp,
-      max: stateObj.attributes.max_temp,
-      sliderType,
-      change: (value) => this._setTemperature(value),
-      drag: (value) => this._dragEvent(value),
-      value: sliderValue,
-      disabled: sliderValue === null,
-      step: this._stepSize,
-    });
     this._updateSetTemp(uiValue);
   }
 
-  private _genSliderValue(
-    stateObj: ClimateEntity
-  ): [string | number | null, string, string] {
-    let sliderType: string;
-    let sliderValue: string | number | null;
-    let uiValue: string;
-
-    if (stateObj.state === "unavailable") {
-      sliderType = "min-range";
-      sliderValue = null;
-      uiValue = this.hass!.localize("state.default.unavailable");
-    } else if (
-      stateObj.attributes.target_temp_low &&
-      stateObj.attributes.target_temp_high
-    ) {
-      sliderType = "range";
-      sliderValue = `${stateObj.attributes.target_temp_low}, ${
-        stateObj.attributes.target_temp_high
-      }`;
-      uiValue = this.formatTemp(
-        [
-          String(stateObj.attributes.target_temp_low),
-          String(stateObj.attributes.target_temp_high),
-        ],
-        false
-      );
-    } else {
-      sliderType = "min-range";
-      sliderValue = Number.isFinite(Number(stateObj.attributes.temperature))
-        ? stateObj.attributes.temperature
-        : null;
-      uiValue = sliderValue !== null ? String(sliderValue) : "";
+  private get _stepSize(): number {
+    const stateObj = this.hass!.states[this._config!.entity] as ClimateEntity;
+
+    if (stateObj.attributes.target_temp_step) {
+      return stateObj.attributes.target_temp_step;
     }
+    return this.hass!.config.unit_system.temperature === UNIT_F ? 1 : 0.5;
+  }
 
-    return [sliderValue, uiValue, sliderType];
+  private _getSetTemp(stateObj: HassEntity) {
+    return stateObj.state === "unavailable"
+      ? this.hass!.localize("state.default.unavailable")
+      : stateObj.attributes.target_temp_low &&
+        stateObj.attributes.target_temp_high
+      ? this.formatTemp(
+          [
+            stateObj.attributes.target_temp_low,
+            stateObj.attributes.target_temp_high,
+          ],
+          false
+        )
+      : Number.isFinite(Number(stateObj.attributes.temperature))
+      ? stateObj.attributes.temperature
+      : "";
```

Comment:
This is a mess. Let the compiler do this, and make it readable instad. You'll have to maintain this some day.